xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-ui/src/main/java/org/xwiki/test/ui/po/AttachmentsPane.java

```diff
@@ -103,32 +129,41 @@ public void deleteAttachmentByFileByName(String attachmentName)
     {
         // We initialize before so we can remove the animation before the modal is shown
         this.confirmDelete = new ConfirmationModal(By.id("deleteAttachment"));
-        getDriver().findElement(By.xpath("//div[@id='attachmentscontent']//a[text()='" + attachmentName
-            + "']/../../div[contains(@class, 'xwikibuttonlinks')]/a[contains(@class,'deletelink')]")).click();
+        int attachmentsNumber = getNumberOfAttachments();
+        this.attachmentsLivetable.getCell(getAttachmentRowNb(attachmentName), 7)
+            .findElement(By.className("actiondelete")).click();
         this.confirmDelete.clickOk();
-        getDriver().waitUntilElementDisappears(
-            By.xpath("//div[@id='attachmentscontent']//a[text()='" + attachmentName + "']"));
-        getDriver().waitUntilElementIsVisible(By.xpath("//div[@id='Attachmentspane']"));
+
+        getDriver().waitUntilCondition(new ExpectedCondition<Boolean>()
+        {
+            @Override
+            public Boolean apply(WebDriver driver)
+            {
+                return getNumberOfAttachments() == attachmentsNumber - 1;
+            }
+        });
     }
 
     /**
      * Deletes the first attachment.
      */
     public void deleteFirstAttachment()
     {
-        String tmp = getDriver()
-            .findElement(By.xpath("//div[@id='_attachments']/*[1]/div[@class='information']/span[@class='name']"))
-            .getText();
-        getDriver()
-            .findElement(By
-                .xpath("//div[@id='attachmentscontent']//a[text()='" + tmp + "']/../../span[2]/a[@class='deletelink']"))
-            .click();
+        // We initialize before so we can remove the animation before the modal is shown
+        this.confirmDelete = new ConfirmationModal(By.id("deleteAttachment"));
+        int attachmentsNumber = getNumberOfAttachments();
+        this.attachmentsLivetable.getCell(1, 7).findElement(By.className("actiondelete")).click();
+        this.confirmDelete.clickOk();
+
+        getDriver().waitUntilCondition(new ExpectedCondition<Boolean>()
+        {
+            @Override
+            public Boolean apply(WebDriver driver)
+            {
+                return getNumberOfAttachments() == attachmentsNumber - 1;
```

Comment:
So from what I see you don't rely on the livetable update but on the counter of attachment update. Couldn't that be a source of flicker in the future if the counter is updated but the LT is still refreshing?